### PR TITLE
Announcing Scala.js 1.4.0.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
Ping @gzm0

File: _posts/news/2021-01-12-announcing-scalajs-1.4.0.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    Some(x.asInstanceOf[T])
+  else
+    None
+}
+{% endhighlight %}
+
+## Dynamic module loading
+
+In Scala.js 1.3.0, we had introduced [module splitting support]({{ BASE_PATH }}/doc/project/module.html#module-splitting).
+Scala.js 1.4.0 goes one step further with dynamic module loading.
```
Comment (gzm0) [inline _posts/news/2021-01-12-announcing-scalajs-1.4.0.md:84]: Might be worth mentioning that dynamic module loading works with any splitting mode and with any number of entry points.

File: _posts/news/2021-01-12-announcing-scalajs-1.4.0.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+}
+{% endhighlight %}
+
+Previously, if the `runtimeClass` was `java.lang.Integer`, and the value of `x` was the integer `5`, the above code would successfully return `Some(5)`.
+With the fixed `isAssignableFrom`, it will return `None` on Scala.js 1.4.0.
+
+This happens because `x.getClass()` returns the smallest numeric type that can hold the value `5`, i.e., `java.lang.Byte`.
+The previous implementation of `isAssignableFrom` would return `true` for `classOf[j.l.Integer].isAssignableFrom(classOf[j.l.Byte])`, but it now returns `false`.
+
+The above idiom can be fixed by advantageously replacing `isAssignableFrom` by `isInstance`:
```
Comment (gzm0) [inline _posts/news/2021-01-12-announcing-scalajs-1.4.0.md:70]: I'm a bit thrown off by the usage of "advantageously" here. Would simply omitting it be acceptable?

File: _posts/news/2021-01-12-announcing-scalajs-1.4.0.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+This is a **minor** release:
+
+* It is backward binary compatible with all earlier versions in the 1.x series: libraries compiled with 1.0.x through 1.3.x can be used with 1.4.0 without change.
+* It is *not* forward binary compatible with 1.3.x: libraries compiled with 1.4.0 cannot be used with 1.3.x or earlier.
+* It is *not* entirely backward source compatible: it is not guaranteed that a codebase will compile *as is* when upgrading from 1.3.x (in particular in the presence of `-Xfatal-warnings`).
+
+As a reminder, libraries compiled with 0.6.x cannot be used with Scala.js 1.x; they must be republished with 1.x first.
+
+## Fixes with compatibility concerns
```
Comment (gzm0) [inline _posts/news/2021-01-12-announcing-scalajs-1.4.0.md:45]: I wonder if the following need mentioning (probably not, just want to point it out):

https://github.com/scala-js/scala-js/commit/0b51d7bdc93a5df738f4c09aabf03bbb6f862685
https://github.com/scala-js/scala-js/issues/4212

**Review by gzm0 — APPROVED** [review]
Only minor things.

File: _posts/news/2021-01-12-announcing-scalajs-1.4.0.md (hunk 3 above)
Comment (sjrd) [reply _posts/news/2021-01-12-announcing-scalajs-1.4.0.md:45]: No, I don't think so. I've added https://github.com/scala-js/scala-js/issues/4212 to the list of issues at the end, though.

File: _posts/news/2021-01-12-announcing-scalajs-1.4.0.md (hunk 2 above)
Comment (sjrd) [reply _posts/news/2021-01-12-announcing-scalajs-1.4.0.md]: Removed.

File: _posts/news/2021-01-12-announcing-scalajs-1.4.0.md (hunk 1 above)
Comment (sjrd) [reply _posts/news/2021-01-12-announcing-scalajs-1.4.0.md:84]: Good point. I added a sentence at the end of the paragraph.

**Review by sjrd — COMMENTED** [review]
Thanks for the review :)

**Pull request merged**